openedx/core/djangoapps/content_tagging/utils.py

```diff
@@ -42,3 +46,57 @@ def get_context_key_from_key_string(key_str: str) -> CourseKey | LibraryLocatorV
         return context_key
 
     raise ValueError("context must be a CourseKey or a LibraryLocatorV2")
+
+
+class TaggingRulesCache:
+    """
+    Caches data required for computing rules for the duration of the request.
```

Comment:
This may not have been necessary if we'd have used `bridgekeeper` instead of `rules` for defining the permissions. My bad for that original mistake. It's a nice solution for now though :)